peer.go

```diff
@@ -159,11 +163,62 @@ func (l *PeerList) Copy() map[string]*Peer {
 
 	listCopy := make(map[string]*Peer)
 	for k, v := range l.peersByHostPort {
-		listCopy[k] = v
+		listCopy[k] = v.Peer
 	}
 	return listCopy
 }
 
+// exists checks if a hostport exists in the peer list.
+func (l *PeerList) exists(hostPort string) (*peerScore, bool) {
+	l.RLock()
+	ps, ok := l.peersByHostPort[hostPort]
+	l.RUnlock()
+
+	return ps, ok
+}
+
+// UpdatePeerHeap updates the peer heap.
+func (l *PeerList) UpdatePeerHeap(p *Peer) {
+
+	ps, ok := l.exists(p.hostPort)
+	if !ok {
+		return
+	}
+
+	score := l.scoreCalculator.GetScore(p)
+	ps.RLock()
+	if score == ps.score {
+		ps.RUnlock()
+		return
+	}
+	ps.RUnlock()
+
+	l.Lock()
+	l.peerHeap.RemovePeer(ps)
```

Comment:
Instead of removing the peer and readding it, can we update the peer using the `heap.Fix`? it's supposed to be more efficient